rusoto/signature/src/payload.rs

```diff
@@ -0,0 +1,56 @@
+// todo: move to hyper::Payload
+use bytes::Bytes;
+use futures::{Async, Poll, Stream};
+use hyper::body::Payload;
+use std::io;
+
+use crate::SignedRequestPayload;
+
+pub struct Body {
```

Comment:
This was basically what exists today as `HttpClientPayload` rebranded as `Body`. If I can figure out `hyper::Body` to type check with `SignedRequestPayload` types, this can go away entirely. It's currently a shim to get this pull in to a compiling state

Reply:
I addressed this here: https://github.com/softprops/rusoto/pull/1